django/contrib/gis/db/backends/spatialite/operations.py

```diff
@@ -149,19 +163,17 @@ def geo_db_type(self, f):
     def get_distance(self, f, value, lookup_type, **kwargs):
         """
         Returns the distance parameters for the given geometry field,
-        lookup value, and lookup type.  SpatiaLite only supports regular
-        cartesian-based queries (no spheroid/sphere calculations for point
-        geometries like PostGIS).
+        lookup value, and lookup type.
         """
         if not value:
             return []
         value = value[0]
         if isinstance(value, Distance):
             if f.geodetic(self.connection):
-                raise ValueError('SpatiaLite does not support distance queries on '
-                                 'geometry fields with a geodetic coordinate system. '
-                                 'Distance objects; use a numeric value of your '
-                                 'distance in degrees instead.')
+                if lookup_type == 'dwithin':
+                    raise ValueError('Only numeric values of degree units are '
```

Comment:
It's tested [here](https://github.com/django/django/blob/9992decbf11d33c83b9ffe58e477ccb99ad51559/tests/gis_tests/distapp/tests.py#L79).